### PR TITLE
docstring

### LINKED ISSUE #356: Check for incomplete docstrings and complete them
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

In many files docstrings are incomplete, some does not describe parameters and return values. 
### Expected Behaviour:

Every docstring must describe all parameters, return value and raised exceptions of the function being documented.
### Do you want to work on this issue?

No

### PR DESCRIPTION
**Pull request by thrishik7**
Documented docstring with parameters and return value for functions in helper.py, resources.py, app_factory.py 

Fixes #356 

### Checklist
- [ ] My branch is up-to-date with upstream/develop branch.
- [ ] Everything works and tested for Python 3.5.2 and above.

### CODE REVIEW

**Comment by xadahiya** [comment]
Build is failing. @thrishik7 please fix this

**Comment by thrishik7** [comment]
Okay . I will look on it

**Comment by thrishik7** [comment]
@xadahiya 
  I just added docstrings, how does it cause an issue on building.

**Comment by xadahiya** [comment]
Your docstrings are not PEP8 compliant. See the build logs

**Comment by thrishik7** [comment]
> Your docstrings are not PEP8 compliant. See the build logs

fine @xadahiya , I will work on it

**Comment by thrishik7** [comment]
@xadahiya , why is it taking to long to build?

**Comment by chrizandr** [comment]
@thrishik7 The builds have failed. You can click on details to see the build info, sometimes it does not update properly on Github. https://travis-ci.com/HTTP-APIs/hydrus/builds/146743379

**Comment by thrishik7** [comment]
> @thrishik7 The builds have failed. You can click on details to see the build info, sometimes it does not update properly on Github. https://travis-ci.com/HTTP-APIs/hydrus/builds/146743379

@chrizandr , okay I will look on it.

**Comment by thrishik7** [comment]
I dont know why Im getting this error </home/travis/build/HTTP-APIs/hydrus/hydrus/resources.py:515:101: E501 line too long (111 > 100 characters)>

Screenshot from 2020-02-10 15-38-41

Can someone please help me out?

File: hydrus/resources.py [hunk 1]
```diff
@@ -513,8 +512,8 @@ def put(self, path, int_list="") -> Response:
                     if not check_required_props(class_path, obj):
                         incomplete_objects.append(obj)
                         object_.remove(obj)
-                link_props_list, link_type_check = get_link_props_for_multiple_objects(class_path,
-                                                                                       object_)
+                link_props_list, link_type_check = get_link_props_for_multiple_objects(class_path, object_)
```
Comment (sameshl) [inline hydrus/resources.py]: @thrishik7 You are getting the `E501 line too long (111 > 100 characters)` from here. Try to undo this change to fix this error.

Comment (thrishik7) [reply hydrus/resources.py]: okay @sameshl I will try.

**Comment by thrishik7** [comment]
@sameshl  @xadahiya  @chrizandr  built successfully by python 3.5. But failing with other version of python.
`ModuleNotFoundError: No module named 'pip'
The command "pip install -r requirements.txt --no-cache" failed and exited with 1 during .`

It seems pip is not installed

**Comment by chrizandr** [comment]
Weird. Try rebuilding? This has never happened before. @xadahiya

**Comment by xadahiya** [comment]
This is because virtualenv 20 was released yesterday. The PR is fine, the issue is on the travis side. See this
https://travis-ci.community/t/python-development-versions-no-longer-include-pip-due-to-virtualenv-20-x/7180/10

**Comment by thrishik7** [comment]
@xadahiya  @chrizandr Can you please review my PR .

File: hydrus/itemhelpers.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    get_link_props)
+from hydrus.utils import (
+    get_session,
+    get_api_name,
+    get_hydrus_server_url)
+from hydrus.socketio_factory import socketio
+
+
+def items_get_check_support(id_, class_type, class_path, path):
+    """Check if class_type supports GET operation"""
```
Comment (xadahiya) [inline hydrus/itemhelpers.py]: You should add docstrings in a format similar to `helpers.py` (see the file above).
